manager/dispatcher/nodes.go

```diff
@@ -86,8 +89,13 @@ func (s *nodeStore) CheckRateLimit(id string) error {
 	s.mu.Lock()
 	defer s.mu.Unlock()
 	if existRn, ok := s.nodes[id]; ok {
+		existRn.Attempts++
 		if time.Since(existRn.Registered) < s.rateLimitPeriod {
-			return grpc.Errorf(codes.Unavailable, "node %s attempted registration too recently", id)
+			if existRn.Attempts > rateLimitCount {
+				return grpc.Errorf(codes.Unavailable, "node %s attempted registration too recently", id)
+			}
+		} else {
+			existRn.Attempts = 0
```

Comment:
0 is more consistent with `Add()` function because the first attempt is 0 from the beginning. I think it can be interpreted as the `kth` attempt. But then `rateLimitCount` should be re-interpreted according to `existRn.Attempts > rateLimitCount`.